test/parallel/test-repl-IIFE.js

```diff
@@ -0,0 +1,54 @@
+'use strict';
+
+// REPL regression test for IIFE https://github.com/nodejs/node/issues/38685
+
+require('../common');
+const util = require('util');
+const assert = require('assert');
+const ArrayStream = require('../common/arraystream');
+const repl = require('repl');
+
+// Create a dummy stream that does nothing
+const stream = new ArrayStream();
+let output = '';
+
+function testNormalIIFE() {
+  const server = runRepl();
+
+  stream.run(['(() => true)()']);
+  assert.strictEqual(output, '> true\n> ');
+  output = '';
+  server.close();
+  testAsyncIIEF();
+}
+
+function testAsyncIIEF() {
+  const server = runRepl();
+  const asyncFn = '(async() => {' +
+    'await new Promise((resolve, reject) => {' +
+      'resolve(true)' +
+    '});' +
+  '})()';
```

Comment:
This could be simplified:
```suggestion
  const asyncFn = '(async() => true)();';
```
or if we want to test with `await`:
```suggestion
  const asyncFn = '(async() => await true)()';
```